exec/java-exec/src/main/java/org/apache/drill/exec/planner/physical/NestedLoopJoinPrule.java

```diff
@@ -49,7 +49,7 @@ protected boolean checkPreconditions(DrillJoinRel join, RelNode left, RelNode ri
       PlannerSettings settings) {
     JoinRelType type = join.getJoinType();
 
-    if (! (type == JoinRelType.INNER || type == JoinRelType.LEFT)) {
+    if (!(type == JoinRelType.INNER || (type == JoinRelType.LEFT && JoinUtils.hasScalarSubqueryInput(left, right)))) {
```

Comment:
I see.  So, Calcite is converting a query with a subquery into an equivalent left join with a scalar subquery.  Do you know if the query returns the correct result in that case if a nested loop join is used?  If not, then it seems like we should still be returning an error.

Also, how do you distinguish the case where the query has been converted to this form by Calcite vs a user explicitly doing a left outer join to a scalar subquery.  It seems like in this case, if Drill uses a nested loop join, it will also return wrong result.